packages/react-components/src/components/MessageThread.test.tsx

```diff
@@ -225,3 +225,290 @@ describe('Message should display inline image correctly', () => {
+    messages[0].content = message1ContentAfterEdit;
+    messages[0].editedOn = new Date('2019-04-13T00:01:00.000+08:10');
+    const expectedOnRenderMentionCount = 2;
+    let onRenderMentionCount = 0;
+
+    rerender(
+      <MessageThread
+        userId={user2Id}
+        messages={messages}
+        mentionOptions={{
+          displayOptions: {
+            onRenderMention: (mention, defaultOnMentionRender) => {
+              onRenderMentionCount++;
+              return <span key={Math.random().toString()}>{defaultOnMentionRender(mention)}</span>;
+            }
+          }
+        }}
+      />
+    );
+
+    expect(onRenderMentionCount).toEqual(expectedOnRenderMentionCount);
+    expect(container.querySelector(`#${user2Id}`)?.nodeName.toLowerCase()).toEqual('msft-mention');
+    expect(container.querySelector(`#${user2Id}`)?.textContent).toEqual(user2Name);
+    expect(container.querySelector(`#${user3Id}`)?.nodeName.toLowerCase()).toEqual('msft-mention');
+    expect(container.querySelector(`#${user3Id}`)?.textContent).toEqual(user3Name);
+  });
+
+  test('Edit Message with @ will show MentionPopover and mentions in edited message', async () => {
+    const user1Id = 'user1';
+    const user2Id = 'user2';
+
+    const user1Name = 'Kat Larsson';
+
+    const messages: ChatMessage[] = [
+      {
+        messageType: 'chat',
+        senderId: user2Id,
+        senderDisplayName: 'Robert Tolbert',
+        messageId: Math.random().toString(),
+        content: 'Absolutely! What seems to be the problem?',
+        createdOn: new Date('2019-04-13T00:00:00.000+08:11'),
+        mine: true,
+        attached: false,
+        contentType: 'html'
+      }
+    ];
+
+    const expectedOnUpdateMessageCount = 1;
+    let onUpdateMessageCount = 0;
+    const onUpdateMessageCallback = (messageId, content): Promise<void> => {
+      const msgIdx = messages.findIndex((m) => m.messageId === messageId);
+      const message = messages[msgIdx];
+      message.content = content;
+      message.editedOn = new Date(Date.now());
+      messages[msgIdx] = message;
+      onUpdateMessageCount++;
+      return Promise.resolve();
+    };
+
+    const onQueryUpdated = async (query: string): Promise<Mention[]> => {
+      return Promise.resolve(
+        [
+          {
+            id: user1Id,
+            displayText: user1Name
+          },
+          {
+            id: 'everyone',
+            displayText: 'Everyone'
+          }
+        ].filter((suggestion) => suggestion.displayText.toLocaleLowerCase().startsWith(query.toLocaleLowerCase()))
+      );
+    };
+
+    const { container, rerender } = render(
+      <MessageThread
+        userId={user2Id}
+        messages={messages}
+        onUpdateMessage={onUpdateMessageCallback}
+        mentionOptions={{
+          displayOptions: {
+            onRenderMention: (mention, defaultOnMentionRender) => {
+              return <span key={Math.random().toString()}>{defaultOnMentionRender(mention)}</span>;

```

Comment:
Math.random should be used for randomness. For test we should be fully controlling the keys / IDs to expected values.